product_uom_updatable/models/product_template.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 Acsone SA/NV
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from itertools import groupby
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = "product.template"
+
+    def write(self, vals):
+        uom_id = vals.pop("uom_id", False)
+        uom_po_id = vals.pop("uom_po_id", False)
+        res = super(ProductTemplate, self).write(vals)
+        if uom_id:
+            self._update_uom(uom_id)
+        if uom_po_id:
+            self._update_uom_po(uom_po_id)
+        return res
+
+    def _update_uom(self, uom_id):
+        uom_obj = self.env["uom.uom"]
+        for key, products_group in groupby(self, key=lambda r: r.uom_id):
```

Comment:
When using `groupby`, records have to be sorted using the same key (unless you actually want to have several groups for the same key if they are not consecutive, but not the case here).
From https://docs.python.org/3.7/library/itertools.html#itertools.groupby:

> The operation of groupby() is similar to the uniq filter in Unix. It generates a break or new group every time the value of the key function changes (which is why it is usually necessary to have sorted the data using the same key function). That behavior differs from SQL’s GROUP BY which aggregates common elements regardless of their input order.